Automattic/wp-calypso
# My-Sites: add Paladin route

Adds a new top-level route at `/paladin/:site`

This is a slimmed-down version of #4429 

This is an attempt to re-create the Customizer as a native Calypso component rather than an iframe for wp-admin. The resulting tool should be significantly faster than the wp-admin version (which has the loading time of the customizer plus the loading time of the iframe) as well as much easier to modify (since it won't be conflicting with changes to the Core customizer).

Furthermore, we'll be able to use it to experiment with and improve upon the Customizer experience (eg: content editing, home page settings) at a pace which has been unavailable when working on the core Customizer. This is basically the same reason that Calypso was first introduced.

Also also, using native calypso tools we'll be able to run the Customizer easily from the desktop app as well as re-use existing Calypso components to make the experience feel more integrated.

Finally, we can add new components that can be used in other parts of Calypso to modify site settings from places other than the Customizer.

<img width="714" alt="screen shot 2016-08-16 at 6 06 53 pm" src="https://cloud.githubusercontent.com/assets/2036909/17717495/485a7254-63dc-11e6-9a81-6e67512ba361.png">

This will need some refactoring when #7477 lands (or the other way around).

# Testing

First, make sure regular previews still work: click the site preview button in the my-sites view and make sure the site preview shows up correctly.

<img width="250" alt="screen_shot_2016-08-16_at_4_32_06_pm" src="https://cloud.githubusercontent.com/assets/2036909/17863814/4fd6cac8-6869-11e6-9532-c83f391a1cb3.png">

To try out the new route, visit `/paladin` and pick a site. You should see the preview appear with a sidebar (See screenshot above). 

1. Using the sidebar you should be able to change the site title and tagline of your site.
2. Changing the site title and tagline should modify both the preview *and* the site card at the top of the sidebar.
3. Try pressing "← Back" at the top of the sidebar to exit the preview. It should warn you that you will lose unsaved changes. Press cancel.
4. Press "Save" to save your changes. In a new window, visit the front-end of your site to verify that the changes have been saved.
5. Press "← Back" again at the top of the sidebar to exit the preview. This time no warning should appear, and the preview should disappear. You should end up on the stats page for your site.
6. Resize the viewport to ~300px and make sure that the sidebar gains a "Preview Changes" button and the preview gets a "Edit" button that each toggle the other.

![paladin-mobile-preview](https://cloud.githubusercontent.com/assets/2036909/18057287/58b0fb7c-6ddd-11e6-80a2-a539b4719ed8.gif)


Test live: https://calypso.live/?branch=add/paladin-preview
